### PR TITLE
Add TEPs requirements to pull request template checklist

### PR DESCRIPTION
**Pull request by jerop**
# Changes

Any user-facing enhancements in Tekton Pipelines need to be approved in
an implementable Tekton Enhancement Proposal.

Today, contributors may not know this requirement when they open pull
requests needing reviewers to catch it. Also, user-facing PRs can also
be mistakenly merged when the TEPs are not yet in implementable state.

In API WG on 03/07/2022, we discussed adding this requirement to the PR
checklist so that contributors are aware of it, or as a reminder.

References:
- https://github.com/tektoncd/community/tree/main/teps
- https://github.com/tektoncd/community/blob/main/teps/0001-tekton-enhancement-proposal-process.md

/kind misc

cc @bobcatfish 

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [n/a] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [n/a] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by lbernick** [comment]
I like the idea of adding this item to the PR checklist, but I would change the wording of "user facing changes". For example, a bug fix is a user facing change but doesn't require a TEP. maybe link to ["what type of work should be tracked in a TEP"](https://github.com/tektoncd/community/blob/main/teps/0001-tekton-enhancement-proposal-process.md#what-type-of-work-should-be-tracked-by-a-tep).

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -21,6 +21,9 @@ As the author of this PR, please check off the items in this checklist:
 - [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
 - [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
   functionality, content, code)
+- [ ] User-facing changes have a [Tekton Enhancement Proposal](https://github.com/tektoncd/community/tree/main/teps) in an
+  [*implementable*](https://github.com/tektoncd/community/blob/main/teps/0001-tekton-enhancement-proposal-process.md#tep-workflow)
+  state that is referenced in the [changes](#changes) section.
```
Comment (pritidesai) [inline .github/pull_request_template.md:26]: How about: 

are the changes part of any TEP? If so, is the TEP approved and in `implementable` state?

Comment (jerop) [reply .github/pull_request_template.md]: this is simpler, but excludes about the part of what needs to be in a TEP

**Comment by jerop** [comment]
@lbernick good idea to clarify which kinds of changes - updated it to `enhancements` and linked the doc specifying what should be in a TEP

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -16,11 +16,11 @@ Supported kinds are: bug, cleanup, design, documentation, feature, flake, misc,
...
-- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
-- [ ] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
-- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
-- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
-  functionality, content, code)
+- [ ] [Docs][docs] included if any changes are user facing
+- [ ] [Tests][tests] included if any functionality added or changed
+- [ ] Follows the [commit message standard][commits]
+- [ ] Meets the [Tekton contributor standards][standards] (including functionality, content, code)
+- [ ] User-facing [enhancements][enhancement] have a [Tekton Enhancement Proposal][teps] that's [*implementable*][tep-workflow]
```
[the remark this answers is not in the document]
Comment (jerop) [reply .github/pull_request_template.md:23]: Most of that is defined in the linked TEP-0001: https://github.com/tektoncd/community/blob/main/teps/0001-tekton-enhancement-proposal-process.md#what-type-of-work-should-be-tracked-by-a-tep

"The definition of what constitutes an "enhancement" is a foundational concern for the Tekton project. Roughly any **Tekton user or operator facing enhancement should follow the TEP process**. If an enhancement would be described in either written or verbal communication to anyone besides the TEP author or developer, then consider creating a TEP. This means any change that may impact any other community project in a way should be proposed as a TEP." 

That link also provides examples of what constitutes a user-facing enhancement that should be in a TEP, and what does not 

> Its fairly simple to check if the changes I am implementing are the result/part of any TEP? If so, is the TEP approved and in `implementable`?

Agreed, but would like to express that certain changes in a PR need a TEP in the first place, do you have thoughts on how to do that effectively? Or do you prefer that Owners review PRs and make that decision?

[the remark this answers is not in the document]
Comment (jerop) [reply .github/pull_request_template.md:23]: also happy to keep it simple as you've suggested if we can't find a way to define "what should be in a TEP"

**Pull request closed without merging**